flake.nix

```diff
@@ -0,0 +1,31 @@
+{
+  description = "NixOS on the Steam Deck";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+  };
+
+  outputs = { self, nixpkgs }: let
+    supportedSystems = [ "x86_64-linux" ];
+    eachSupportedSystem = f: nixpkgs.lib.genAttrs supportedSystems (system: let
+      pkgs = import nixpkgs {
+        inherit system;
+        config.allowUnfree = true;
+        overlays = [ self.overlays.default ];
+      };
+    in f pkgs);
+  in {
+    legacyPackages = eachSupportedSystem (pkgs: pkgs);
+
+    nixosModules = rec {
+      default = jovian;
+      jovian = ./modules;
+    };
```

Comment:
Question, about flake stuff: is using that flake modules output going to intrinsically use the pinned Nixpkgs?

Reply:
You are right that it won't be affected by our `nixpkgs`.